FWCore/Framework/src/TransitionProcessors.icc

```diff
@@ -19,9 +19,14 @@ struct FileResources {
   ~FileResources() {
     // See the message in catch clause
     CMS_SA_ALLOW try {
-      ep_.respondToCloseInputFile();
-      ep_.closeInputFile(cleaningUpAfterException_);
-      ep_.closeOutputFiles();
+      // Don't try to execute the following sequence of functions twice.
+      // If the sequence was already attempted and failed, then do nothing.
+      if (!closingSequenceAlreadyFailed_) {
```

Comment:
What prompted this change?

Reply:
I was testing by intentionally throwing exceptions in different transitions to see that the behavior after an exception and the printed messages were OK. When I tried this in endProcessBlock under certain circumstances it would throw an exception and then later try running the endProcessBlock transition again in the FileResources destructor and terminate immediately instead of cleanly dealing with the exception and printing the intended message.